lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -327,36 +344,42 @@ namespace Intl
         );
     }
 
-    HRESULT SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
+    void SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
     {
         // We know what actual type we stored in the IPlatformAgnosticResource*, so cast to it.
-        auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
-        icu::NumberFormat *nf = numFormatResource->GetInstance();
-        // REVIEW (doilij): Without RTTI support, we can't do dynamic_cast,
-        // which is the preferred way to know whether it was indeed created as a DecimalFormat,
-        // which is not something the API will indicate to us directly (create returns a NumberFormat which may or may not be a DecimalFormat)
-        icu::DecimalFormat *df = reinterpret_cast<icu::DecimalFormat *>(nf);
-        df->setMinimumSignificantDigits(minSigDigits);
-        df->setMaximumSignificantDigits(maxSigDigits);
-        return S_OK;
+        icu::NumberFormat *nf = UNWRAP_RESOURCE(resource, icu::NumberFormat);
+
+        // TODO(jahorto): Determine if we could ever have a NumberFormat that isn't a DecimalFormat (and if so, what to do here)
```

Comment:
It sounds like we've determined this via the else clause below?

Reply:
Well, the `else { assert }` is just to allow us to hopefully find out _when_ its not the case, but I still don't know what the proper resolution to that scenario is.

Reply:
Personally I think continuing with a less-than-optimal format output is fine. The assert to give us some visibility on when/if it happens will be helpful.